mh/database.py

```diff
@@ -293,16 +298,23 @@ def reserve(self, reserve):
 
 class Gate(object):
     LAYER_DEPTH = 2
+    MAX_CITY_PER_GATE = 40
 
-    def __init__(self, name, parent, city_count=40, player_capacity=100):
+    def __init__(self, name, parent, capacity=100, is_jap=False):
         self.name = name
         self.parent = parent
         self.state = LayerState.EMPTY
+
+        city_max_capacity = City.get_max_capacity(is_jap)
+        city_count = int(min(
+            math.ceil(capacity / city_max_capacity), Gate.MAX_CITY_PER_GATE))
+        capacity = city_count * city_max_capacity
```

Comment:
Won't this approach fail when many users decide to solo play (and not fill up the city)?

Reply:
yes, but even in that case there is not much that we can do, because there is a maximum amount of city per gate.